doc/nrf/protocols/matter/getting_started/memory_profiling.rst

```diff
@@ -0,0 +1,371 @@
+.. _ug_matter_device_memory_profiling:
+
+Profiling memory in Matter applications
+#######################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This guide offers useful commands and Kconfig options for configuring memory sizes for stacks, heaps, and Zephyr settings.
+You can learn to measure memory usage and adjust these values according to your project's requirements.
+
+Measuring memory usage
+======================
+
+You can use several predefined Kconfig options and UART shell commands to obtain the current memory statistics from the device.
+To do this, you can enable the global Matter memory profiling Kconfig, which also activates all other necessary Kconfig options.
+Alternatively, you can enable each option separately.
+
+The Matter memory profiling Kconfig option is a predefined setting that enables all UART shell commands for measuring memory usage.
+To enable it, set the :kconfig:option:`CONFIG_CHIP_MEMORY_PROFILING` Kconfig option to ``y``.
+
+This Kconfig option enables the following functionalities on the Matter device:
+
+- :ref:`ug_matter_configuring_settings_shell` by setting the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_SETTINGS_SHELL` Kconfig option to ``y``.
+- :doc:`matter:nrfconnect_examples_cli` by setting the :kconfig:option:`CONFIG_CHIP_LIB_SHELL` Kconfig option to ``y``.
+- Zephyr Kernel commands by setting the :kconfig:option:`CONFIG_KERNEL_SHELL` Kconfig option to ``y``.
+- Zephyr Settings shell by setting the :kconfig:option:`CONFIG_SETTINGS_SHELL` Kconfig option to ``y``.
+- OpenThread shell by setting the :kconfig:option:`CONFIG_OPENTHREAD_SHELL` Kconfig option to ``y`` if you build the Matter over Thread variant.
+- Chip statistics shell by setting the :kconfig:option:`CONFIG_CHIP_STATISTICS` Kconfig option to ``y``.
+
+All the functionalities listed below are automatically enabled if the Matter memory profiling Kconfig option is activated.
```

Comment:
Shouldn't be above instead of below? Also we have the same information mentioned few lines before:

> This Kconfig option enables the following functionalities on the Matter device:

and 

> Alternatively, you can enable each option separately.